### PR TITLE
Remote build public projects (#80)

### PR DESCRIPTION
**Pull request by beholders-eye**
https://app.clubhouse.io/yourbaseio/story/80/yb-remotebuild-should-work-even-if-project-not-there-yet

Updates the API response expectations; initially supports building public repos (if GH would work)

### CODE REVIEW

File: cli/remote_build.go [hunk 1]
```diff
@@ -780,9 +782,15 @@ func (cmd *RemoteCmd) submitBuild(project *Project, tagMap map[string]string) er
...
+	case 403:
 		submitErrored()
-		return fmt.Errorf("Invalid data sent to the YB API")
+		if cmd.publicRepo {
+			return fmt.Errorf("This should not happen, please open a support inquery with YB")
+		} else {
+			return fmt.Errorf("Tried to build a private repository of a organization of which you're not part of.\nPlease check with the '%s' admins", project.OrgSlug)
+		}
+	case 412:
+		// TODO Show helpfull message with App URL to fix GH App installation issue
```
Comment (beholders-eye) [inline cli/remote_build.go]: @amyhua I need your help with this.

File: cli/remote_build.go [hunk 2]
```diff
@@ -639,12 +640,12 @@ func (p *RemoteCmd) fetchProject(urls []string) (*Project, GitRemote, error) {
 	}
 
 	if resp.StatusCode != 200 {
-		if resp.StatusCode == 404 {
-			return nil, empty, fmt.Errorf("Is YourBase GitHub App installed?\nPlease check '%v'", ybconfig.CurrentGHAppUrl())
+		if resp.StatusCode == 400 {
+			return nil, empty, fmt.Errorf("This is us, not you, please try again in some minutes.")
```
Comment (herbertmd) [inline cli/remote_build.go]: s/some/a few/

File: cli/remote_build.go [hunk 3]
```diff
@@ -780,9 +782,15 @@ func (cmd *RemoteCmd) submitBuild(project *Project, tagMap map[string]string) er
...
 		submitErrored()
 		return fmt.Errorf("Unauthorized, authentication failed.\nPlease `yb login` again.")
-	case 400:
+	case 403:
 		submitErrored()
-		return fmt.Errorf("Invalid data sent to the YB API")
+		if cmd.publicRepo {
+			return fmt.Errorf("This should not happen, please open a support inquery with YB")
+		} else {
+			return fmt.Errorf("Tried to build a private repository of a organization of which you're not part of.\nPlease check with the '%s' admins", project.OrgSlug)
```
Comment (herbertmd) [inline cli/remote_build.go]: I would remove the second sentence so we're not revealing organization details (even slug) with an unauthorized user.

File: cli/remote_build.go [hunk 4]
```diff
@@ -860,6 +868,9 @@ func (cmd *RemoteCmd) submitBuild(project *Project, tagMap map[string]string) er
 						endTime := time.Now()
 						setupTime := endTime.Sub(startTime)
 						log.Infof("Set up finished at %s, taking %s", endTime.Format(TIME_FORMAT), setupTime.Truncate(time.Millisecond))
+						if cmd.publicRepo {
+							log.Warnf("We doesn't support GH integration of a public repo, like: '%s'", project.Repository)
```
Comment (herbertmd) [inline cli/remote_build.go]: s/doesn't/don't/

File: cli/remote_build.go (hunk 2 above)
Comment (beholders-eye) [reply cli/remote_build.go]: Thanks!

**Comment by beholders-eye** [comment]
@johnewart, for me this is good to go.

File: cli/remote_build.go (hunk 1 above)
Comment (amyhua) [reply cli/remote_build.go]: it'd be nice if we returned error type enums (versus specific messages) when throwing errors, so the frontend can iterate on the messaging as needed for the user experience. E.g.
ErrorType.ProviderAppNotInstalled,
ErrorType.ProviderAppInstallRepoNotSelected,
ErrorType.OrgMembershipMissing,
ErrorType.PublicReposNotSupported

**Comment by beholders-eye** [comment]
> @johnewart, IMO this is good to go.

I still think this is good to go, but it depends on yourbase/api#47, that I just finished updating

**Comment by beholders-eye** [comment]
Fixed the message when remote building a public repository. We shouldn't report remote build status to GH...

**Comment by beholders-eye** [comment]
PTAL

**Comment by beholders-eye** [comment]
> PTAL

@herbertmd, could you?

**Pull request merged**